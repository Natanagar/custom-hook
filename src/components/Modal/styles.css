.modal {
  position: relative;
  width: 600px;
  max-width: 100%;

  height: 80px;
  max-height: 100%;

  border: black 1px solid;
  border-radius: 5px;

  margin-top: 15px;
}

.close-button {
  position: absolute;
  top: 0px;
  right: 0px;
  border: none;

  background-color: #34363a;
  width: 1rem;
  height: 1rem;
  padding: 0;
  border: 0;
  cursor: pointer;
  outline: 0;

  color: white;
}

.callback-button {
  background-color: darkred;
  border: black 1px solid;
  border-radius: 10px;
  padding: 10px;
  margin: 2px;
  cursor: pointer;
  color: white;
}
