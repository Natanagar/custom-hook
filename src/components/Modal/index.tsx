import React from "react";
import "./styles.css";

interface Props {
  content: React.ReactElement | string;
  closeModal: (id: string) => void;
  id: string;
  cb: (e: React.MouseEvent<HTMLButtonElement, MouseEvent>, id: string) => void;
}

const Modal: React.FC<Props> = ({ content, closeModal, id, cb }) => {
  return (
    <div className="modal">
      <button
        type="button"
        className="close-button"
        onClick={() => closeModal(id)}
        id={id}
      >
        X
      </button>
      <div>{content}</div>
      <div>
        <button onClick={(e) => cb(e, id)} className="callback-button">
          выполнить колбек
        </button>
      </div>
    </div>
  );
};

export default Modal;
