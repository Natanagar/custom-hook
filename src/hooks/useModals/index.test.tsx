import * as React from "react";
import renderer from "react-test-renderer";
import Modal from "../../components/Modal";

const mockMath = Object.create(global.Math);
mockMath.random = () => 0.5;
global.Math = mockMath;

describe("Modal renders without mistakes", () => {
  it("matches snapshot", () => {
    const testProps = {
      id: "123",
      content: "123",
      cb: () => jest.fn(),
    };

    const tree = renderer.create(<Modal {...testProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
