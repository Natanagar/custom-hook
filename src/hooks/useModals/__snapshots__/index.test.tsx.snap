// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal renders without mistakes matches snapshot 1`] = `
<div
  className="modal"
>
  <button
    className="close-button"
    id="123"
    onClick={[Function]}
    type="button"
  >
    X
  </button>
  <div>
    123
  </div>
  <div>
    <button
      className="callback-button"
      onClick={[Function]}
    >
      выполнить колбек
    </button>
  </div>
</div>
`;
